CMakeLists.txt

```diff
@@ -1,11 +1,9 @@
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.0.2)
```

Comment:
You're right @lexicalunit. I just realized there is a `COMPILE_DEFINITIONS` property for targets in version 2.8.7, which could be used instead of `target_compile_definitions` (from 3.0.2). I'update the PR to reflect this change and some other fixes. Thanks!